lib/transforms/inst_simplify.cc

```diff
@@ -2381,6 +2381,28 @@ std::pair<Def, Def> InstSimplify::RunOnInstruction(SliceInst* inst) {
   return {orig_def, orig_def};
 }
 
+std::pair<Def, Def> InstSimplify::RunOnInstruction(SelectInst* inst) {
+  Def orig_def{inst, 0};
+  auto cond = DynCast<Constant>(inst->GetOperand(0));
+  if (cond == nullptr) {
+    return {orig_def, orig_def};
+  }
+  if (cond->HasSameValueOf(1)) {
+    return {orig_def, inst->GetOperand(1)};
```

Comment:
Shape broadcasting might be needed before returning either the true or false operand.

Reply:
good point